File: specex_pybind/py/specex/myargs.py
# Licensed under a 3-clause BSD style license - see LICENSE.rst
# -*- coding: utf-8 -*-
"""
specex.myargs
=====================

Functions for testing argument passing to c++ functions

"""
from __future__ import absolute_import, division, print_function

import numpy as np

import fitsio

from ._internal import (MYARGS_INT, MyArgs)

def str_to_myargs_type(input):
    if input == "domyargs":
        return MYARGS_INT
    else:
        raise ValueError("unknown target type '{}'".format(input))
    return None

def test_myargs(list_of_args=[]):
    ma = MyArgs(MYARGS_INT)

    ma.get_args(list_of_args)
    ma.print_args()
    
    return None

def test_mymath(a,b):
    ma = MyMath(MYMATH_INT)

    ma.get_args(list_of_args)
    ma.print_args()
    
    return None

File: src/myargs.cpp
// Licensed under a 3-clause BSD style license - see LICENSE.rst

#include <myargs.h>

#include <sstream>

namespace spx = specex;

spx::MyArgs::MyArgs(int id){
  
  this->ids = id;

  return;
}

void spx::MyArgs::get_args(int argc, char **argv){
  std::vector<std::string> argvc(argv, argv + argc);
  this->argc = argc;
  this->argv = argvc;

  return;
}

void spx::MyArgs::print_args(){
  std::cout << "id in print_args = " << this->ids << std::endl;
  for (auto iarg = 0; iarg < this->argc; iarg++)
    std::cout << "arg " << iarg << " is: "
	      << this->argv[iarg] << std::endl;
    
  return;
}

